libraries/app/database_api.cpp

```diff
@@ -1554,11 +1554,17 @@ vector<market_trade> database_api_impl::get_trade_history( const string& base,
          {
             trade.sequence = -itr->key.sequence;
             trade.side1_account_id = itr->op.account_id;
-            trade.side1_was_selling = itr->op.pays.asset_id == assets[0]->id ? base : quote;
+            if(itr->op.receives.asset_id == assets[0]->id)
+               trade.type = "buy";
+            else
+               trade.type = "sell";
```

Comment:
The code seems wrong. The correct logic should be `the maker receives some base asset == the taker sells some base asset == the trade is a proactive sell`. For example, in the EUR/USD market, if Alice put up an order buying EUR (the base asset), she is the buyer, she expects to receive some quote asset (USD), if she is the maker, it means the opponent sold some EUR into her position, so it's a proactive sell.